src/gc.c

```diff
@@ -1196,6 +1196,16 @@ static NOINLINE jl_taggedvalue_t *add_page(jl_gc_pool_t *p) JL_NOTSAFEPOINT
     return fl;
 }
 
+// called into by LLVM-generated code
+JL_DLLEXPORT jl_value_t *jl_gc_pool_alloc_outer(jl_ptls_t ptls, int pool_offset,
```

Comment:
I would prefer keeping the same name here, and renaming the inner `internal` implementation, which then also shouldn't be exported (and ideally inlined into this function).

Maybe call that one then `jl_gc_pool_alloc_impl`

Reply:
We talked about inlining it too, but we weren't sure about how to inline it _only_ into this function..

If we mark that one inline, it'll end up inlining into the normal callers of it, though.. So the _C_ functions (indicated in Pete's graph, above) that calls `jl_gc_pool_alloc` would now call `jl_gc_pool_alloc_impl` instead, and then it would be inlined.

So it'd become inlined into `jl_gc_alloc_` and `jl_gc_alloc_string`. Maybe that's okay?

Or would you want us to create yet a third separate wrapper function that gets called by those? So we would have three functions:
- `jl_gc_pool_alloc_from_generated`: called from generated code
- `jl_gc_pool_alloc`: called from `jl_gc_alloc_` and `jl_gc_alloc_string`
- `jl_gc_pool_alloc_impl`: called from `jl_gc_pool_alloc`
? Or is that too weird? And the extra inlining is actually fine?